lib/pdk/cli/exec.rb

```diff
@@ -137,31 +142,15 @@ def execute!
             # Make sure invocation of Ruby prefers our private installation.
             @process.environment['PATH'] = [RbConfig::CONFIG['bindir'], ENV['PATH']].compact.join(File::PATH_SEPARATOR)
 
-            mod_root = PDK::Util.module_root
-
-            unless mod_root
-              @spinner.error
-
-              raise PDK::CLI::FatalError, _('Current working directory is not part of a module. (No metadata.json was found.)')
-            end
-
-            Dir.chdir(mod_root) do
-              ::Bundler.with_clean_env do
-                run_process!
-              end
```

Comment:
We can test `Dir.pwd` so that we only try to `chdir` if we aren't already in the right place. That should keep the child threads from complaining.